### PR TITLE
Basic ES2015 / ES6

### PR DESCRIPTION
**Pull request by alexpate**
Really have no idea what I'm doing when it comes to writing lexers. However have added some support for ES6/ES2016.

By no means a full spec, but it fixes the backticks highlighting error.

[Live example](https://alexpate.uk/journal/pure-svg-progress-circles-javascript-part-2/)

`gem "rouge", :git => 'https://github.com/alexpate/rouge.git'`

### CODE REVIEW

**Comment by jneen** [comment]
Hi @alexpate! Thanks for the fix. Could you add some sample code to `spec/visual/samples/javascript` that exercises the new features you've added? That way we can be sure we don't break them in the future.

**Comment by iRath96** [comment]
I think that template strings are highlighted incorrectly in this PR. The are highlighted as ordinary code, while they should actually be highlighted as strings with only the `${}` portions being highlighted as code. I've implemented the correct behavior for template strings in PR #504.

File: spec/visual/samples/javascript [hunk 1]
```diff
@@ -149,3 +149,7 @@ function* range(from, to)
     yield from++;
   }
 }
+
+// Template Strings - ES2015
+var b = "world";
+console.log(`hello ${b}`);
```
Comment (graingert) [inline spec/visual/samples/javascript]: missing new line

**Comment by naomiajacobs** [comment]
@alexpate thanks for writing this! This is exactly what I need! Do you know if there are any plans to merge this soon?

**Comment by alexpate** [comment]
It looks like this may already have been implemented in #504. See comment above.

**Pull request closed without merging**